plugins/org/src/components/Cards/OwnershipCard/useGetEntities.ts

```diff
@@ -57,78 +57,130 @@ const getQueryParams = (
   return qs.stringify({ filters }, { arrayFormat: 'repeat' });
 };
 
-const getOwnersEntityRef = (owner: Entity): string[] => {
-  let owners = [stringifyEntityRef(owner)];
-  if (owner.kind === 'User') {
-    const ownerGroups = getEntityRelations(owner, RELATION_MEMBER_OF, {
-      kind: 'Group',
-    });
-    const ownerGroupsName = ownerGroups.map(ownerGroup =>
-      stringifyEntityRef({
-        kind: ownerGroup.kind,
-        namespace: ownerGroup.namespace,
-        name: ownerGroup.name,
-      }),
-    );
-    owners = [...owners, ...ownerGroupsName];
+const getMemberOfEntityRefs = (owner: Entity): string[] => {
+  const parentGroups = getEntityRelations(owner, RELATION_MEMBER_OF, {
+    kind: 'Group',
+  });
+
+  const ownerGroupsNames = parentGroups.map(({ kind, namespace, name }) =>
+    stringifyEntityRef({
+      kind,
+      namespace,
+      name,
+    }),
+  );
+
+  const {
+    kind,
+    metadata: { namespace, name },
+  } = owner;
+
+  return [...ownerGroupsNames, stringifyEntityRef({ kind, namespace, name })];
+};
+
+const isEntity = (entity: Entity | undefined): entity is Entity =>
+  entity !== undefined;
+
+const getChildOwnershipEntityRefs = async (
+  entity: Entity,
+  catalogApi: CatalogApi,
+): Promise<string[]> => {
+  const childGroups = getEntityRelations(entity, RELATION_PARENT_OF, {
+    kind: 'Group',
+  });
+
+  const hasChildGroups = childGroups.length > 0;
+
+  if (hasChildGroups) {
+    const childGroupEntities = (
+      await Promise.all(
+        childGroups.map(childGroup =>
+          limiter(() => catalogApi.getEntityByRef(childGroup)),
+        ),
+      )
+    ).filter(isEntity);
+
+    return (
+      await Promise.all(
+        childGroupEntities.map(childGroupEntity =>
+          limiter(() =>
+            getChildOwnershipEntityRefs(childGroupEntity, catalogApi),
+          ),
+        ),
+      )
+    ).flatMap(aggregated => aggregated);
   }
-  return owners;
+
+  return [
+    stringifyEntityRef({
+      kind: entity.kind,
+      namespace: entity.metadata.namespace,
+      name: entity.metadata.name,
+    }),
+  ];
 };
 
-const getAggregatedOwnersEntityRef = async (
```

Comment:
Moved this one to recursive solution for better readability. See `getChildOwnershipEntityRefs`